src/history/HistoryTests.cpp

```diff
@@ -204,6 +206,168 @@ TEST_CASE("History publish queueing", "[history][historydelay][historycatchup]")
           catchupSimulation.getApp().getLedgerManager().getLedgerNum());
 }
 
+TEST_CASE("History bucket verification",
+          "[history][bucketverification[batching]]")
+{
+    /* Tests bucket verification stage of catchup. Assumes ledger chain
+     * verification was successful. **/
+
+    CatchupSimulation catchupSimulation{};
```

Comment:
None of these perform catchup end to end, "History bucket verification" specifically tests DownloadBucketsWork. `catchupSimulation` is needed to publish correct history and get proper hashes. The second app does not really start catchup, it spins up a test app with proper configs (which perhaps could be changed to simply using TestApplication).

Reply:
right, my point is exactly that: because those tests don't perform end to end catchup, we don't need the test infrastructure that is used for end to end catchup (the overkill part), especially because here you are actually overriding the data produced anyways.